bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/provider/file/AbstractFileProvider.java

```diff
@@ -317,7 +317,7 @@ private String getParserType(URL url) {
             return Parser.FORMAT_JSON;
         }
         String fileExtension = fileName.substring(fileName.lastIndexOf(".") + 1);
```

Comment:
Store `fileName.lastIndexOf(".")` in a local variable earlier and reuse it here.

Reply:
Done! And I threw the Elvis operator in the mix. ?:-)